crates/analytics/src/auth_events/core.rs

```diff
@@ -0,0 +1,115 @@
+use std::collections::HashMap;
+
+use api_models::analytics::{
+    auth_events::{AuthEventMetrics, AuthEventMetricsBucketIdentifier, MetricsBucketResponse},
+    AnalyticsMetadata, GetAuthEventMetricRequest, MetricsResponse,
+};
+use error_stack::ResultExt;
+use router_env::{instrument, logger, tracing};
+
+use super::AuthEventMetricsAccumulator;
+use crate::{
+    auth_events::AuthEventMetricAccumulator,
+    errors::{AnalyticsError, AnalyticsResult},
+    AnalyticsProvider,
+};
+
+#[instrument(skip_all)]
+pub async fn get_metrics(
+    pool: &AnalyticsProvider,
+    merchant_id: &String,
+    publishable_key: Option<&String>,
+    req: GetAuthEventMetricRequest,
+) -> AnalyticsResult<MetricsResponse<MetricsBucketResponse>> {
+    let mut metrics_accumulator: HashMap<
+        AuthEventMetricsBucketIdentifier,
+        AuthEventMetricsAccumulator,
+    > = HashMap::new();
+
+    if let Some(publishable_key) = publishable_key {
+        let mut set = tokio::task::JoinSet::new();
+        for metric_type in req.metrics.iter().cloned() {
+            let req = req.clone();
+            let merchant_id_scoped = merchant_id.to_owned();
+            let publishable_key_scoped = publishable_key.to_owned();
+            let pool = pool.clone();
+            set.spawn(async move {
+                let data = pool
+                    .get_auth_event_metrics(
+                        &metric_type,
+                        &merchant_id_scoped,
+                        &publishable_key_scoped,
+                        &req.time_series.map(|t| t.granularity),
+                        &req.time_range,
+                    )
+                    .await
+                    .change_context(AnalyticsError::UnknownError);
+                (metric_type, data)
+            });
+        }
+
+        while let Some((metric, data)) = set
+            .join_next()
+            .await
+            .transpose()
+            .change_context(AnalyticsError::UnknownError)?
+        {
+            logger::info!("Logging Result {:?}", data);
```

Comment:
do we need this logger?

if we do add more context here as to which metric was this data for etc.

Reply:
Removed the logger